# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/MediaEngineWebRTC.h
# Commit: 1248d2f659a9
# Full Hash: 1248d2f659a9f341f0de5d6558eecf364fc6f0a2
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-03-29 21:53:01
# Regressor Bug: 1494675
# File Overlap Count: 1
# Description:
#   Bug 1494675 - Remove windowID to MediaEngineSource mappings from MediaEngines. r=padenot
#   
#   These once served a caching-and-reuse purpose it seems, but it makes less sense
#   when sources are not shared.
#   
# ==============================================================================

diff -r 20543d03a4cd -r 1248d2f659a9 dom/media/webrtc/MediaEngineWebRTC.h
--- a/dom/media/webrtc/MediaEngineWebRTC.h	Wed Mar 27 14:10:24 2019 +0000
+++ b/dom/media/webrtc/MediaEngineWebRTC.h	Wed Mar 27 14:10:10 2019 +0000
@@ -64,7 +64,6 @@
 
   void EnumerateDevices(uint64_t aWindowId, dom::MediaSourceEnum, MediaSinkEnum,
                         nsTArray<RefPtr<MediaDevice>>*) override;
-  void ReleaseResourcesForWindow(uint64_t aWindowId) override;
 
  private:
   ~MediaEngineWebRTC() = default;
@@ -84,15 +83,6 @@
   // This also is set in the ctor and then never changed, but we can't make it
   // const because we pass it to a function that takes bool* in the ctor.
   bool mHasTabVideoSource;
-
-  // Maps WindowID to a map of device uuid to their MediaEngineSource,
-  // separately for audio and video.
-  nsClassHashtable<nsUint64HashKey,
-                   nsRefPtrHashtable<nsStringHashKey, MediaEngineSource>>
-      mVideoSources;
-  nsClassHashtable<nsUint64HashKey,
-                   nsRefPtrHashtable<nsStringHashKey, MediaEngineSource>>
-      mAudioSources;
 };
 
 }  // namespace mozilla
